87.md

```diff
@@ -0,0 +1,129 @@
+The first value of the tag is the `kind:38173`/`kind:38172` event identifier, the second value of the tag is a relay hint, and the third value of the tag is the type of ecash mint (currently `cashu` or `fedimint`).
+
+## Ecash Mint Information
+
+Cashu mints SHOULD publish `kind:38172` events to announce their capabilities and how to connect to them.
+
+For cashu mints, the `u` tag SHOULD be the URL to the cashu mint and it should list the nuts that the cashu mint supports.
+
+The `d` tag SHOULD be the mint's pubkey (found when querying `/v1/info`), this way users can query by pubkey and find the mint announcement.
+
+```json
+{
+  "kind": 38172,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <cashu mint pubkey>],
+    ["u", "https://cashu.example.com"],
+    ["nuts", "1,2,3,4,5,6,7"]
+  ]
+}
+```
+
+Fedimints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
+
+For fedimints, it should list all known fedimint invite codes in `u` tags and it should list the modules it supports.
+
+The `d` tag SHOULD be the federation id, this way users can query by federation id and find the fedimint announcement.
+
+```json
+{
+  "kind": 38173,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <federation-id>],
+    ["u", "fed11abc.."],
+    ["u", "fed11xyz.."],
+    ["modules", "lightning,wallet,mint"]
+  ]
+}
+```
+
+* `content` is an optional `metadata`-like stringified JSON object, as described in NIP-01. This content is useful when the pubkey creating the `kind:38173` is not a normal user. If `content` is empty, the `kind:0` of the pubkey should be used to display mint information (e.g. name, picture, web, LUD16, etc.)
+
+## Example
+
+### User A recommends some mints
+User A might be a user of a cashu mint. Using a client, user A publishes an event recommending the cashu mint they use.
+
+```json
+{
+  "kind": 18173,
+  "tags": [
+    ["u", "fed11abc..", "fedimint"],
+    ["u", "https://cashu.example.com", "cashu"],
+    ["a", "38173:fedimint-pubkey:<d-identifier>", "wss://relay1", "fedimint"],
+    ["a", "38172:cashu-mint-pubkey:<d-identifier>", "wss://relay2", "cashu"]
+  ],
+  ...
+}
+```
+
+### User B finds a mint
+User B wants to use an ecash wallet, they need to find a mint.
+
+User B's wallet client queries for `kind:18173` events, looking for recommendations for ecash mints.
+
+```json
+["REQ", <id>, [{ "kinds": [18173], "authors": [<user>, <users-contact-list>] }]]
+```
+
+User B, who follows User A, sees that `kind:18173` event and tries to connect to the corresponding mints.
+
+### Alternative query bypassing `kind:18173`
+Alternatively, users might choose to query directly for `kind:38173` for an event kind. Clients SHOULD be careful doing this and use spam-prevention mechanisms or querying high-quality restricted relays to avoid directing users to malicious handlers.
```

Comment:
Would it make sense for `kind:38173` to contain some kind of attestation from the mint custodians to independently verify its authenticity? 

Could be optional if it's a hassle for some mints.

Reply:
I think this is doable for cashu, they can just use the same key for ecash to sign the nostr event (might just need to invert for xonly stuff). For fedimints I'm not sure if there is an easy way to prove ownership, maybe signing with the BLS keys, however this could end up being a ton of signatures.

cc @elsirion

Reply:
There's little sense in signing imo. The federation id commits to the federation config. So even if the API URLs get spoofed the attacker can only:

* Return valid config with different API endpoints that will be used after that
* Return invalid config which will be rejected

So the only thing an attacker can achieve is making you ask more API endpoints for the correct config til one returns it.

Reply:
Just joining a federation one doesn't know is a horrible idea anyway, so I only expect this query to be used when the (short) federation id is known but no API URL. That could be useful to make e-cash notes smaller while still being able to claim notes from unknown federations.